consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java

```diff
@@ -0,0 +1,281 @@
+import org.hyperledger.besu.crypto.NodeKey;
+import org.hyperledger.besu.crypto.SECP256K1.Signature;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.chain.MinedBlockObserver;
+import org.hyperledger.besu.ethereum.core.Block;
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+import org.hyperledger.besu.ethereum.core.BlockImporter;
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.ethereum.mainnet.HeaderValidationMode;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModuleException;
+import org.hyperledger.besu.util.Subscribers;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class QbftRound {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Subscribers<MinedBlockObserver> observers;
+  private final RoundState roundState;
+  private final BftBlockCreator blockCreator;
+  private final ProtocolContext protocolContext;
+  private final BlockImporter blockImporter;
+  private final NodeKey nodeKey;
+  private final MessageFactory messageFactory; // used only to create stored local msgs
+  private final QbftMessageTransmitter transmitter;
+
+  public QbftRound(
+      final RoundState roundState,
+      final BftBlockCreator blockCreator,
+      final ProtocolContext protocolContext,
+      final BlockImporter blockImporter,
+      final Subscribers<MinedBlockObserver> observers,
+      final NodeKey nodeKey,
+      final MessageFactory messageFactory,
+      final QbftMessageTransmitter transmitter,
+      final RoundTimer roundTimer) {
+    this.roundState = roundState;
+    this.blockCreator = blockCreator;
+    this.protocolContext = protocolContext;
+    this.blockImporter = blockImporter;
+    this.observers = observers;
+    this.nodeKey = nodeKey;
+    this.messageFactory = messageFactory;
+    this.transmitter = transmitter;
+
+    roundTimer.startTimer(getRoundIdentifier());
+  }
+
+  public ConsensusRoundIdentifier getRoundIdentifier() {
+    return roundState.getRoundIdentifier();
+  }
+
+  public void createAndSendProposalMessage(final long headerTimeStampSeconds) {
+    final Block block = blockCreator.createBlock(headerTimeStampSeconds);
+    final BftExtraData extraData = BftExtraData.decode(block.getHeader());
+    LOG.debug("Creating proposed block. round={}", roundState.getRoundIdentifier());
+    LOG.trace(
+        "Creating proposed block with extraData={} blockHeader={}", extraData, block.getHeader());
+    updateStateWithProposalAndTransmit(block, emptyList(), emptyList());
+  }
+
+  public void startRoundWith(
+      final RoundChangeMetadata roundChangeMetadata, final long headerTimestamp) {
+    final Optional<Block> bestBlockFromRoundChange = roundChangeMetadata.getBlock();
+    Block blockToPublish;
+    if (!bestBlockFromRoundChange.isPresent()) {
```

Comment:
can simplify to bestBlockFromRoundChange.isEmpty()

Reply:
done